tollway/utils.py

```diff
@@ -1,43 +1,70 @@
 import json
+import logging
+from collections import namedtuple
 from pathlib import Path
-from typing import Mapping, TypedDict, Union
+from typing import Callable, Mapping, TypedDict, Union
 
 from decouple import config
 from google.cloud import pubsub_v1
+from google.cloud.pubsub_v1.publisher.futures import Future
 from google.oauth2 import service_account
 
+Topic = namedtuple("Topic", ["publisher", "topic_path"])
+
 
 class EventsLog(TypedDict):
     late_events: dict[str, list[str]]
     past_events: list[dict[str, str]]
     all_events: list[Mapping[str, Union[str, bool]]]
 
 
-def get_topic(pubsub: bool) -> tuple:
+def get_topic(pubsub: bool) -> Topic:
     if pubsub:
         project = config("PROJECT_ID")
         topic = config("TOPIC_ID")
         credentials = service_account.Credentials.from_service_account_file(config("PUBSUB_SERVICE_ACCOUNT"))
         publisher = pubsub_v1.PublisherClient(credentials=credentials)
         topic_path = publisher.topic_path(project=project, topic=topic)
-        return publisher, topic_path
-    return None, None
+        return Topic(publisher=publisher, topic_path=topic_path)
+    return Topic(publisher=None, topic_path=None)
+
+
+def get_pubsub_logger(pubsub: bool) -> logging.Logger:
+    logger = logging.getLogger("pubsub")
+    if pubsub:
+        logging.basicConfig(
```

Comment:
maybe set this up at the module or even package (__init__.py) level?

Reply:
I like this idea